lib/dest/write-contents/write-buffer.js

```diff
@@ -1,14 +1,30 @@
 'use strict';
 
+var iconv = require('iconv-lite');
+
 var fo = require('../../file-operations');
+var getCodec = require('../../codecs');
 
 function writeBuffer(file, optResolver, onWritten) {
+  var encoding = optResolver.resolve('encoding', file);
+  var codec = getCodec(encoding);
```

Comment:
Do you imagine folks providing their own implementation of iconv's codec API, as the value to the `encoding` option here? I guess that might work, with a few more changes, but seems a pretty far-fetched scenario... Much more likely they would just submit their codec to iconv, right?

Also, notice that we only use this codec object for two properties, namely `enc` and `bomAware`. The public API of iconv expects its arguments to `encode` and `decode` to be strings.